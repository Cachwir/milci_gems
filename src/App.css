.switch-wrapper {
  display: grid;
  grid-template-columns: 100%;
}

.switch-wrapper > div {
  grid-row-start: 1;
  grid-row-end: 1;
  grid-column-start: 1;
  grid-column-end: 1;
}

.container.floated {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.container .floated {
  float: none !important;
  display: block;
  position: relative !important;
}

@media only screen and (max-width: 767px){
  .about .container.floated {
    display: block;
  }
}
