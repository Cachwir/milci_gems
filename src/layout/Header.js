import React from "react";
import {Link} from "react-router-dom";

function Header()
{
    return (
        <div>
            <div id="top-line"></div>

            <div className="container">

                {/* Header */}
                <header id="header">
                    {/* Logo */}
                    <div className="ten columns">
                        <div id="logo">
                            <h1><Link to="/"><img src={process.env.PUBLIC_URL + "/assets/images/logo.png"} alt="logo" /></Link></h1>
                            <div id="tagline">Des gemmes d'exception</div>
                            <div className="clearfix"></div>
                        </div>
                    </div>

                    {/* Social / Contact */}
                    <div className="six columns">

                        <div className="clearfix"></div>

                    </div>
                </header>

                <div className="clearfix"></div>

            </div>
        </div>
    );
}

export default Header;