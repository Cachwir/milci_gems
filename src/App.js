import React, {Component} from "react";

import "./App.css";
import Header from "./layout/Header";
import Menu from "./layout/Menu";
import Footer from "./layout/Footer";
import AppRouter from "./AppRouter";
import {BrowserRouter as Router} from "react-router-dom";

class App extends Component {

  /**
   * L'appli sera composée de trois pages :
   * - une page de garde
   * - une gallerie
   * - à propos (pour expliquer comment ce site a été fait, dans quel contexte et ajouter des remerciements et des
   * sources)
   */

  routerEvents = [];

  registerRouterEvent(event)
  {
      this.routerEvents.push(event);
  }

  render() {
      return (
          <Router>
              <div id="wrapper">
                  <Header />
                  <Menu Parent={this} />
                  <AppRouter onRouteChangeEvents={this.routerEvents}  />
                  <Footer />
              </div>
          </Router>
      );
  }
}

export default App;
