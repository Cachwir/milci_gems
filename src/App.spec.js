import { shallow } from 'enzyme'
import { expect } from 'chai'
import React from 'react'
import App from './App'
import sinon from "sinon";
import Header from "./layout/Header";

describe('<App />', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).to.contain(<Header />);
    });

    it('should match its reference snapshot', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).to.matchSnapshot();
    });
})