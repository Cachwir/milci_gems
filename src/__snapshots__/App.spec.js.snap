// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match its reference snapshot 1`] = `
<BrowserRouter>
  <div
    id="wrapper"
  >
    <Header />
    <withRouter(MenuWithoutRouter)
      Parent={
        App {
          "context": Object {},
          "props": Object {},
          "refs": Object {},
          "routerEvents": Array [],
          "setState": [Function],
          "state": null,
          "updater": Updater {
            "_callbacks": Array [],
            "_renderer": ReactShallowRenderer {
              "_context": Object {},
              "_didScheduleRenderPhaseUpdate": false,
              "_dispatcher": Object {
                "readContext": [Function],
                "useCallback": [Function],
                "useContext": [Function],
                "useDebugValue": [Function],
                "useDeferredValue": [Function],
                "useEffect": [Function],
                "useImperativeHandle": [Function],
                "useLayoutEffect": [Function],
                "useMemo": [Function],
                "useReducer": [Function],
                "useRef": [Function],
                "useResponder": [Function],
                "useState": [Function],
                "useTransition": [Function],
              },
              "_element": <App />,
              "_firstWorkInProgressHook": null,
              "_forcedUpdate": false,
              "_instance": [Circular],
              "_isReRender": false,
              "_newState": null,
              "_numberOfReRenders": 0,
              "_renderPhaseUpdates": null,
              "_rendered": <BrowserRouter>
                <div
                  id="wrapper"
                >
                  <Header />
                  <withRouter(MenuWithoutRouter)
                    Parent={[Circular]}
                  />
                  <withRouter(AppRouterWithoutRouter)
                    onRouteChangeEvents={Array []}
                  />
                  <Footer />
                </div>
              </BrowserRouter>,
              "_rendering": false,
              "_updater": [Circular],
              "_workInProgressHook": null,
            },
          },
        }
      }
    />
    <withRouter(AppRouterWithoutRouter)
      onRouteChangeEvents={Array []}
    />
    <Footer />
  </div>
</BrowserRouter>
`;
